### ISSUE TITLE
Error in qr.default(t(V)) : NA/NaN/Inf in foreign function call (arg 1)

### ISSUE THREAD

**Issue by nachocab** [drisso/archive-zinbwave#35]
I was able to run zinbwave in the past, but with the attached dataset (1000 rows x 153 columns) I'm getting an error:

```

zinbwave::zinbFit(temp, K = 2, epsilon = 1000, ncores = 4)
# Error in qr.default(t(V)) : NA/NaN/Inf in foreign function call (arg 1)
```

An idea what it could be?

[temp.txt](https://github.com/drisso/zinbwave/files/985073/temp.txt)

**Comment by drisso** [comment]
Hi @nachocab 
the zinbwave package is based on a negative binomial distribution, which assumes that your input is count data. In your case, the temp matrix has non-integer values, hence the error.

This works for me:

```{r}
res <- zinbwave::zinbFit(round(temp), K = 2, epsilon = 1000, ncores = 4)
```

We should probably throw an error when the input is non-integer values. I will add an issue to fix that, thanks for letting us know!

**Comment by drisso** [comment]
Should be addressed by issue #36

**Comment by nachocab** [comment]
Thanks for the quick response! I didn't think to check for whole counts (I was pretty sure it was), so I think a specific warning about this would be useful 👍
